MariaDB 10.1.34

Mno-hime opened the pull request:
Update MariaDB to 10.1.34.

Removed `test/` as the tests seems to be run other way. Tests are
dumping cores for both i386 & amd64 but that's not a regression:

```
 Thread 21 (LWP    4        ):
 #0  0xfffffd7ffad4e387 in __lwp_park () from /lib/64/libc.so.1
 #1  0xfffffd7ffad476f9 in cond_sleep_queue () from /lib/64/libc.so.1
 #2  0xfffffd7ffad478fb in cond_wait_queue () from /lib/64/libc.so.1
 #3  0xfffffd7ffad47fb3 in __cond_wait () from /lib/64/libc.so.1
 #4  0xfffffd7ffad47fea in cond_wait () from /lib/64/libc.so.1
 #5  0xfffffd7ffad48025 in pthread_cond_wait () from /lib/64/libc.so.1
 #6  0xfffffd7ff6b8bfc9 in os_event_wait_low(os_event*, long) () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/amd64/mysql-test/var/2/plugins/ha_innodb.so
 #7  0xfffffd7ff6b8a468 in os_aio_simulated_handle(unsigned long, fil_node_t**, void**, unsigned long*) () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/amd64/mysql-test/var/2/plugins/ha_innodb.so
 #8  0xfffffd7ff6ad99b0 in fil_aio_wait(unsigned long) () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/amd64/mysql-test/var/2/plugins/ha_innodb.so
 #9  0xfffffd7ff6c042b8 in io_handler_thread () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/amd64/mysql-test/var/2/plugins/ha_innodb.so
 #10 0xfffffd7ffad4e02a in _thrp_setup () from /lib/64/libc.so.1
 #11 0xfffffd7ffad4e340 in ?? () from /lib/64/libc.so.1
 #12 0x0000000000000000 in ?? ()
``

Replaced `_sun` with `__sun` in 01-cfi-issue.patch.

Tested: DB start, password set, information retrieval.

pyhalov commented on components/database/mariadb-101/Makefile:
```diff
@@ -73,16 +77,44 @@ CMAKE_OPTIONS+= -DWITH_SSL=system
 CMAKE_OPTIONS+= -DCMAKE_REQUIRED_INCLUDES=/usr/include/pcre
+CMAKE_OPTIONS+= -DWITH_INNOBASE_STORAGE_ENGINE=1
+CMAKE_OPTIONS+= -DWITH_INNODB_MEMCACHED=ON
+CMAKE_OPTIONS+= -DWITH_INNODB_DISALLOW_WRITES=1
```
Do we really need this? This seems to be something Galera-related.

Mno-hime committed:
MariaDB 10.1.34

Update MariaDB to 10.1.34.

Removed `test/` as the tests seems to be run other way. Tests are
dumping cores for both i386 & amd64 but that's not a regression:

```
 Thread 21 (LWP    4        ):
 #0  0xfffffd7ffad4e387 in __lwp_park () from /lib/64/libc.so.1
 #1  0xfffffd7ffad476f9 in cond_sleep_queue () from /lib/64/libc.so.1
 #2  0xfffffd7ffad478fb in cond_wait_queue () from /lib/64/libc.so.1
 #3  0xfffffd7ffad47fb3 in __cond_wait () from /lib/64/libc.so.1
 #4  0xfffffd7ffad47fea in cond_wait () from /lib/64/libc.so.1
 #5  0xfffffd7ffad48025 in pthread_cond_wait () from /lib/64/libc.so.1
 #6  0xfffffd7ff6b8bfc9 in os_event_wait_low(os_event*, long) () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/amd64/mysql-test/var/2/plugins/ha_innodb.so
 #7  0xfffffd7ff6b8a468 in os_aio_simulated_handle(unsigned long, fil_node_t**, void**, unsigned long*) () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/am
mysql-test/var/2/plugins/ha_innodb.so
 #8  0xfffffd7ff6ad99b0 in fil_aio_wait(unsigned long) () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/amd64/mysql-test/var/2/plugins/ha_innodb.so
 #9  0xfffffd7ff6c042b8 in io_handler_thread () from /export/home/newman/ws/oi-userland/components/database/mariadb-101/build/amd64/mysql-test/var/2/plugins/ha_innodb.so
 #10 0xfffffd7ffad4e02a in _thrp_setup () from /lib/64/libc.so.1
 #11 0xfffffd7ffad4e340 in ?? () from /lib/64/libc.so.1
 #12 0x0000000000000000 in ?? ()
```

Replaced `_sun` with `__sun` in 01-cfi-issue.patch.

Tested: DB start, password set, information retrieval.
